### DISCUSSION TITLE
Pylontech H2

### DISCUSSION THREAD

**Discussion by janchlebek** [mletenay/home-assistant-pylontech#1] [General]
Hi,

I am not really familiar with all the model numbers - the question is: do you know if this works with the Pylontech H2 battery pack/BMS?

Thanks
Jan

**Comment by mletenay** [comment]
I cannot say for sure, but H1/H2 seem to be just more fancy packaged X1/X2 systems.
I expect quite similar hardware beneath it, so it could in theory work and some potential firmware differences could be easily addressed.
You have to try to connect to RS232 port and see if the console can be activated and provides the standard commands.

**Comment by Smart-Home-48** [comment]
Hello, I need help with the setup please. I installed esphome, flashed the Wemos D1 with the firmware, but I don't know which values I have to enter in the "yaml". Please have an example. Greetings Thomas

**Reply by mletenay** [reply]
This integration is configured via UI, there is no yaml config necessary.
Do you mean yaml of the ESPHome configuration of https://github.com/mletenay/esphome-stream-server ?

**Comment by jirkanvck** [comment]
Yesterday I wrote about Pylontech BMS integration (issues). Batteries are Force H2, BMS FC0500MA, MAX3232/Wemos D1 RS232 converter. For Pylontech, there are pylontech, pylonbatt and pylontech-bms variants in ESPhome, I tried all them insert to Integration and they listen on port 6053 (tried initiated from putty raw - "Bad indicator byte" and log - "[W][api.connection:071]: IP: Socket operation failed: BAD_INDICATOR errno=11" . What I can't do is add your Pylontech BMS integration and connect to RS232 converter, it always ends with the error Connection failed. I really appreciate and admire your work, thank you for any help. Jiri

**Reply by mletenay** [reply]
will look at it over weekend ...

**Reply by mletenay** [reply]
... something wrong is happening out there, I lost connect to my ESP as well, still investigating ...

**Reply by mletenay** [reply]
OK, there was some breaking change in HA 2023.4 and I had to update to stream_server upstream to include this fix - https://github.com/oxan/esphome-stream-server/commit/4ef23f01d499f44a707ca9dd838158741d029840
Could you try again if that helped ?

**Comment by jirkanvck** [comment]
Thank you very much for Your investigation, for two days I will try it. Jiri

**Comment by jirkanvck** [comment]
I think it will be something small - the stream server is running in the ESPHome element (pylontech.local:1234), but the attempt to connect the Pylontech BMS integration to this address:port still ends with Failed to connect. What can it be? Thank You Jiri

**Comment by jirkanvck** [comment]
Hello, finally success, Pylontech BMS works. The problems were partly in the HW (connection Wemos - MAX3232) and the missing UART definition in the .yaml stream server. Thank you very much for the excellent integration! Jiri

**Comment by adler00x** [comment]
Hello jirkanvck,
I'm not a software expert and I'm facing several problems with my Pylontech PowerCube X1 hardware configured as 1x SC0500-40 BMS and 10x H48050 Batteries. Could you please provide here your YAML file that works successful. Tank you in advance!
